lib/puppet/provider/package/aix.rb

```diff
@@ -96,6 +105,11 @@ def install(useversion = true)
     if output =~ /^#{Regexp.escape(@resource[:name])}\s+.*\s+Already superseded by.*$/
       self.fail _("aix package provider is unable to downgrade packages")
     end
+
+    status = query[:status]
```

Comment:
I think `query` may return `nil` if `lslpp` execution raises exception and this may lead to `NoMethodError`. https://github.com/puppetlabs/puppet/blob/430f73f13ab616f0910a348ce7a0db38e07bbf64/lib/puppet/provider/package/aix.rb#L130-L132